lightning/src/ln/channel.rs

```diff
@@ -4197,6 +4196,15 @@ impl<Signer: Sign> Channel<Signer> {
 		);
 	}
 
+	/// Allowed in any state (including after shutdown)
+	pub fn get_counterparty_htlc_max_msat(&self) -> u64 {
+		return cmp::min(
+			self.channel_value_satoshis * 1000 * 9 / 10,
```

Comment:
Hmm, I'm not so sure about taking the 90% discount on the `ChannelDetails`, it feels a bit strange to tell users "the htlc max is X" when its really Y. That said, I do think we need to subtract that channel reserve values (both our own and our counterparty's).

Note: same applies for the holder HTLC max above, we probably don't want to change how we announce channels, but we do want to expose it differently in the `ChannelDetails`.

Reply:
So it would look like `self.channel_value_satoshis - self.counterparty_selected_channel_reserve_satoshis - self.holder_selected_chan_reserve_msat`?

Reply:
Yes, with unit conversion. May need an `unwrap_or(0)`